src/api.ts

```diff
@@ -123,16 +147,78 @@ type Poll<T, X> = Variant<{
   wait: Promise<void>
 }>
 
-export interface Aggregate extends Piece {
-  
-}
+export interface Aggregate extends Piece {}
 
 export interface AggregateView extends Aggregate, PieceView {
   indexSize: number
   limit: number
   tree: AggregateTree
+
+  /**
+   * Resolves inclusion proof for the given piece. If aggregate does not include
+   * the given piece `{ error: RangeError }` is returned.
+   */
+  resolveProof(piece: PieceLink): Result<InclusionProofView, RangeError>
 }
 
+/**
+ * @see https://github.com/filecoin-project/go-data-segment/blob/master/datasegment/verifier.go#L8-L14
+ */
+export type AggregationProof = [
+  InclusionProof,
+  AuxDataType,
+  SingletonMarketSource
+]
+
+/**
+ * @see https://github.com/filecoin-project/go-data-segment/blob/master/datasegment/verifier.go#L16-L18
+ */
+export type SingletonMarketSource = [DealID]
+/**
+ * @see https://github.com/filecoin-project/go-state-types/blob/master/abi/deal.go#L5
+ */
+export type DealID = uint64
+
+/**
+ * @see https://github.com/filecoin-project/go-data-segment/blob/master/datasegment/verifier.go#L12
+ */
+export type AuxDataType = 0
+
+/**
+ * This is the encode layout of the `AggregationProof` used in the go
+ * implementation.
+ *
+ * @see https://github.com/filecoin-project/go-data-segment/blob/e3257b64fa2c84e0df95df35de409cfed7a38438/datasegment/inclusion.go#L31-L39
+ */
+export type InclusionProofLayout = [
+  // ProofSubtree is proof of inclusion of the client's data segment in the data aggregator's Merkle tree (includes position information)
+  // I.e. a proof that the root node of the subtree containing all the nodes (leafs) of a data segment is contained in CommDA
+  ProofDataLayout,
+  // ProofIndex is a proof that an entry for the user's data is contained in the index of the aggregator's deal.
+  // I.e. a proof that the data segment index constructed from the root of the user's data segment subtree is contained in the index of the deal tree.
+  ProofDataLayout
+]
```

Comment:
what's going on here? are the comments like ` // ProofSubtree` a clue to what this `ProofDataLayout` is used for? Or are they legacy? Some type aliases might help here.

Reply:
```suggestion
export type InclusionProofLayout = [
  InclusionProof['tree'],
  InclusionProof['index']
]
```

Moving those field descriptions into `InclusionProof` type where they are more appropriate.